src/ui/src/custom_components/core.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * Declare custom components
+ *
+ * @param customComponents list of components in key-value form
+ */
+export function declareCustomComponents(
+	customComponents: Record<string, unknown>,
+) {
+	Object.entries(customComponents).forEach(([key, customComponent]) => {
+		_auditCustomComponents(key, customComponent);
+	});
+
+	return customComponents;
+}
+
+/**
+ * Audit the validity of custom components
+ *
+ * @param key
+ * @param customComponent
+ */
+export function _auditCustomComponents(key: string, customComponent: unknown) {
```

Comment:
1. You can remove the `_` prefix and simply not export this function
2. I think the function name could be more meaningfull
```suggestion
function checkComponentName(key: string) {
```